### PR TITLE
Grab '@param' tags from initializers

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
Addresses @zhengbli's concern in https://github.com/Microsoft/TypeScript/pull/4978#discussion_r40481626.

This PR enables us to grab parameters from a "simple"/"canonical" callable/constructable expression in a variable statement with a single declaration.

We only do this for a function expression, arrow function, or class expression with a constructor, or any of the aforementioned in nested parentheses.

In the presence of a class expression, if there are multiple constructor declarations, the parameters are acquired from the first one.

### CODE REVIEW

File: src/services/services.ts [hunk 1]
```diff
@@ -7096,6 +7096,52 @@ namespace ts {
...
+                    return getParametersFromRightHandSideOfAssignment(varDeclarations[0].initializer);
+                }
+            }
+
+            return emptyArray;
+        }
+
+        /**
+         * Digs into an an initializer or RHS operand of an assignment operation
+         * to get the parameters from an apst signature corresponding to a
```
Comment (DanielRosenwasser) [inline src/services/services.ts]: apt*

**Comment by mhegazy** [comment]
:+1:

**Comment by RyanCavanaugh** [comment]
:+1:

**Comment by vladima** [comment]
:+1:

**Comment by zhengbli** [comment]
:+1:

**Pull request merged**